src/Sylius/Bundle/FixturesBundle/DataFixtures/ORM/LoadProductsData.php

```diff
@@ -261,14 +262,13 @@ protected function generateVariants(ProductInterface $product)
     }
 
     /**
-     * Adds master variant to product.
-     *
      * @param ProductInterface $product
-     * @param string           $sku
+     * @param string $sku
      */
-    protected function addMasterVariant(ProductInterface $product, $sku = null)
+    protected function addVariant(ProductInterface $product, $sku = null)
     {
-        $variant = $product->getMasterVariant();
+        /** @var ProductVariantInterface $variant */
+        $variant = $product->getFirstVariant();
```

Comment:
In `createProduct` I would create Product without Variant and here simply create variant and add it to the Product.
`$variant = $this->variantFactory->createNew();`